tests/test-screenshot.lua

```diff
@@ -0,0 +1,164 @@
+        y = s.geometry.y + 1,
+    })
+
+    -- Top right
+    table.insert(corners, {
+        x = s.geometry.x + s.geometry.width - 11,
+        y = s.geometry.y + 1,
+    })
+
+    -- Bottom left
+    table.insert(corners, {
+        x = s.geometry.x + 1,
+        y = s.geometry.y + s.geometry.height - 11,
+    })
+
+    -- Bottom right
+    table.insert(corners, {
+        x = s.geometry.x + s.geometry.width - 11,
+        y = s.geometry.y + s.geometry.height - 11,
+    })
+
+end
+
+-- Make 10x10 wibox in each corner to eliminate any geometry errors
+for _, corner in ipairs(corners) do
+    corner.bg = "#ff0000"
+    corner.visible = true
+    corner.width = 10
+    corner.height = 10
+
+    wibox(corner)
+end
+
+local function copy_to_image_surface(content, w, h)
+    local sur = gsurface(content)
+
+    local img = cairo.ImageSurface(cairo.Format.RGB24, w, h)
+    local cr = cairo.Context(img)
+
+    cr:set_source_surface(sur)
+    cr:paint()
+    img:finish()
+
+    return img
+end
+
+local function get_pixel(img, x, y)
+    local bytes = gdk.pixbuf_get_from_surface(img, x, y, 1, 1):get_pixels()
+    return "#" .. bytes:gsub('.', function(c) return ('%02x'):format(c:byte()) end)
+end
+
+local steps = {}
+
+-- Check the whole root window.
+table.insert(steps, function()
+    local img = copy_to_image_surface(root.content(), root.size())
+    local root_width, root_height = root.size()
+
+    if get_pixel(img, 100, 100) ~= "#00ff00" then return end
+    if get_pixel(img, 2, 2) ~= "#ff0000" then return end
+
+    assert(get_pixel(img, 100, 100) == "#00ff00")
+    assert(get_pixel(img, 199, 199) == "#00ff00")
+    assert(get_pixel(img, 201, 201) ~= "#00ff00")
+
+    assert(get_pixel(img, 2, 2) == "#ff0000")
+    assert(get_pixel(img, root_width - 2, 2) == "#ff0000")
+    assert(get_pixel(img, 2, root_height - 2) == "#ff0000")
+    assert(get_pixel(img, root_width - 2, root_height - 2) == "#ff0000")
+
+    return true
+end)
+
+-- Check the screen.content
+table.insert(steps, function()
+    for s in screen do
+
+      local geo = s.geometry
+      local img = copy_to_image_surface(s.content, geo.width, geo.height)
+
+      assert(get_pixel(img, geo.x + 4, geo.y + 4) == "#ff0000")
+      assert(get_pixel(img, geo.x + geo.width - 4, geo.y + 4) == "#ff0000")
+      assert(get_pixel(img, geo.x + 4, geo.y + geo.height - 4) == "#ff0000")
+      assert(get_pixel(img, geo.x + geo.width - 4, geo.y + geo.height - 4) == "#ff0000")
+
+    end
+
+    -- Spawn for the client.content test
+    assert(#client.get() == 0)
+    spawn(tiny_client)
+
+    return true
+end)
+
+-- Check the client.content
+table.insert(steps, function()
+    if #client.get() ~= 1 then return end
+    local c = client.get()[1]
+    local geo = c:geometry()
+    local img = copy_to_image_surface(c.content, geo.width, geo.height)
+
+    if get_pixel(img, math.floor(geo.width / 2), math.floor(geo.height / 2)) ~= "#0000ff" then
+        return
+    end
+
+    awesome.kill(c.pid, 9)
```

Comment:
nit: @Elv13 can you add a comment here with what you said in the PR thread? It would prevent to forget why we need to `kill -9` the test client.